src/python/pants/pantsd/pants_daemon.py

```diff
@@ -54,8 +56,11 @@ def create(cls, options_bootstrapper: OptionsBootstrapper) -> PantsDaemon:
             bootstrap_options = options_bootstrapper.bootstrap_options
             bootstrap_options_values = bootstrap_options.for_global_scope()
 
+        # This executor is owned by the PantsDaemon, and borrowed by the Pants runs that are launched by
+        # PantsDaemonCore. Individual runs will call shutdown to tear down the executor, but those calls
+        # have no effect on a borrowed executor.
         executor = GlobalOptions.create_py_executor(bootstrap_options_values)
-        core = PantsDaemonCore(options_bootstrapper, executor, cls._setup_services)
+        core = PantsDaemonCore(options_bootstrapper, executor.to_borrowed(), cls._setup_services)
```

Comment:
It might be a nice bonus in the future to circle back and have `to_borrowed` return a different type such that `PantsDaemonCore` could be type-checked to only ever get a borrowed executor here in Python land.